tools/depcheck/pkg/cmd/trace/filter.go

```diff
@@ -0,0 +1,143 @@
+package trace
+
+import (
+	"strings"
+
+	"github.com/gonum/graph"
+
+	depgraph "github.com/openshift/origin/tools/depcheck/pkg/graph"
+)
+
+// FilterVendorPackages receives a graph and a set of vendorRootNames contained within the graph.
+// Returns a new graph with the sub-tree for each vendorRoot collapsed into just the vendorRoot.
+// Relationships among vendorRoots are kept: if a transitive closure exists between two
+// vendor sub-trees, an edge is created between their roots in the newly collapsed graph.
+func FilterVendorPackages(g *depgraph.MutableDirectedGraph, vendorRootNames []string) (*depgraph.MutableDirectedGraph, error) {
```

Comment:
this isn't actually a vendorRootName. This is a list of "allowedNodes" or "packagePrefix" or something like that.